python/revng/internal/cli/_commands/generate_migrations/__init__.py

```diff
@@ -0,0 +1,239 @@
+        return raw_input
+
+    def _get_raw_input_develop_branch(self, schema_path: str) -> str:
+        try:
+            git_process = subprocess.run(
+                f"git --no-pager show develop:{schema_path}",
+                shell=True,
+                check=True,
+                capture_output=True,
+            )
+        except subprocess.CalledProcessError as error:
+            raise GenerateMigrationException(
+                f"Error while running git: {error}\n{error.stderr.decode().strip()}"
+            )
+
+        raw_schema = git_process.stdout.decode()
+
+        return raw_schema
+
+    def _restructure(self, schema: Dict) -> Dict:
+        """Restructures the schema into a nested dictionaries where the `Name` field of types and
+        fields/members are keys.
+        """
+        result: Dict = {}
+
+        result["version"] = schema["version"] if "version" in schema else None
+        result["definitions"] = {}
+
+        for definition in schema["definitions"]:
+            definition_name = definition["name"]
+
+            sub_definitions_list = (
+                definition["members"] if definition["type"] == "enum" else definition["fields"]
+            )
+
+            sub_definitions = {}
+            for sub_definition in sub_definitions_list:
+                sub_definitions_name = sub_definition["name"]
+                sub_definition_properties = {k: v for k, v in sub_definition.items() if k != "name"}
+
+                sub_definitions[sub_definitions_name] = sub_definition_properties
+            definition_properties = {
+                k: v for k, v in definition.items() if k not in {"name", "fields", "members"}
+            }
+
+            result["definitions"][definition_name] = definition_properties, sub_definitions
+
+        return result
+
+    def _process_schema(self, raw_schema: str) -> Dict:
+        """Parse the schema into YAML, validate the YAML against the metaschema, and restructure."""
+        metaschema = self._get_metaschema()
+
+        try:
+            unstructured_schema = yaml.safe_load(raw_schema)
+        except yaml.YAMLError as error:
+            raise GenerateMigrationException(f"Error while parsing schema YAML: {error}")
+        try:
+            jsonschema.validate(instance=unstructured_schema, schema=metaschema)
+        except jsonschema.ValidationError as error:
+            raise GenerateMigrationException(f"Schema is invalid: {error.message}")
+
+        return self._restructure(unstructured_schema)
+
+    def _changed_keys(self, previous_properties: Dict, current_properties: Dict) -> List[str]:
+        result: List[str] = list(set(previous_properties) ^ set(current_properties))
+        for key in set(previous_properties) & set(current_properties):
+            if previous_properties[key] != current_properties[key]:
+                result.append(key)
+        return result
+
+    Change = (
```

Comment:
I'd move it out of the class. This module itself is an already good namespace for this.